DoxygenRule.cmake

```diff
@@ -103,35 +98,21 @@ if(NOT DOXYGEN_CONFIG_FILE)
   set(DOXYGEN_CONFIG_FILE ${PROJECT_BINARY_DIR}/doc/Doxyfile)
 endif()
 
-add_custom_target(doxygen_install_${PROJECT_NAME}
+add_custom_target(${PROJECT_NAME}-install
   ${CMAKE_COMMAND} -P ${PROJECT_BINARY_DIR}/cmake_install.cmake
-  DEPENDS ${INSTALL_DEPENDS})
-set_target_properties(doxygen_install_${PROJECT_NAME} PROPERTIES
+  DEPENDS ${PROJECT_NAME}-all)
+set_target_properties(${PROJECT_NAME}-install PROPERTIES
   EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER ${PROJECT_NAME}/doxygen)
 
-if(NOT TARGET doxygen_install)
-  add_custom_target(doxygen_install)
-  set_target_properties(doxygen_install PROPERTIES
-    EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER ${PROJECT_NAME}/doxygen)
-endif()
-add_dependencies(doxygen_install doxygen_install_${PROJECT_NAME})
-
-add_custom_target(doxygen_html_${PROJECT_NAME}
+add_custom_target(${PROJECT_NAME}-html
   ${DOXYGEN_EXECUTABLE} ${DOXYGEN_CONFIG_FILE}
   WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/doc
   COMMENT "Generating ${PROJECT_NAME} API documentation using doxygen" VERBATIM
-  DEPENDS doxygen_install_${PROJECT_NAME} project_info_${PROJECT_NAME})
-set_target_properties(doxygen_html_${PROJECT_NAME} PROPERTIES
+  DEPENDS ${PROJECT_NAME}-install project_info_${PROJECT_NAME})
+set_target_properties(${PROJECT_NAME}-html PROPERTIES
   EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER ${PROJECT_NAME}/doxygen)
 
-if(NOT TARGET doxygen_html)
-  add_custom_target(doxygen_html)
-  set_target_properties(doxygen_html PROPERTIES
-    EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER ${PROJECT_NAME}/doxygen)
-endif()
-add_dependencies(doxygen_html doxygen_html_${PROJECT_NAME})
-
-add_custom_target(${PROJECT_NAME}-doxygen DEPENDS doxygen_html_${PROJECT_NAME})
+add_custom_target(${PROJECT_NAME}-doxygen DEPENDS ${PROJECT_NAME}-html)
```

Comment:
The -html and -install targets are internal and undocumented on purpose.

Removed -html target.